wire-runtime/src/test/proto-kotlin/com/squareup/wire/protos/person/Person.kt

```diff
@@ -0,0 +1,116 @@
+import com.squareup.wire.ProtoAdapter
+import com.squareup.wire.ProtoReader
+import com.squareup.wire.ProtoWriter
+import com.squareup.wire.WireEnum
+import com.squareup.wire.internal.Internal
+import com.squareup.wire.kotlin.UnknownFieldsBuilder
+import com.squareup.wire.kotlin.decodeMessage
+import kotlin.Int
+import kotlin.String
+import kotlin.collections.List
+import kotlin.jvm.JvmOverloads
+import okio.ByteString
+
+data class Person @JvmOverloads constructor(
+    val name: String,
+    val id: Int,
+    val email: String? = null,
+    val phone: List<PhoneNumber> = emptyList(),
+    val unknownFields: ByteString = ByteString.EMPTY
+) {
+    object ADAPTER : ProtoAdapter<Person>(FieldEncoding.LENGTH_DELIMITED, Person::class.java) {
+        override fun encodedSize(value: Person): Int = ProtoAdapter.STRING.encodedSizeWithTag(1, value.name) +
+            ProtoAdapter.INT32.encodedSizeWithTag(2, value.id) +
+            ProtoAdapter.STRING.encodedSizeWithTag(3, value.email) +
+            PhoneNumber.ADAPTER.asRepeated().encodedSizeWithTag(4, value.phone) +
+            value.unknownFields.size()
+
+        override fun encode(writer: ProtoWriter, value: Person) {
+            ProtoAdapter.STRING.encodeWithTag(writer, 1, value.name)
+            ProtoAdapter.INT32.encodeWithTag(writer, 2, value.id)
+            ProtoAdapter.STRING.encodeWithTag(writer, 3, value.email)
+            PhoneNumber.ADAPTER.asRepeated().encodeWithTag(writer, 4, value.phone)
+            writer.writeBytes(value.unknownFields)
+        }
+
+        override fun decode(reader: ProtoReader): Person {
+            var name: String? = null
+            var id: Int? = null
+            var email: String? = null
+            var phone = mutableListOf<PhoneNumber>()
+            val unknownFields = reader.decodeMessage { tag ->
+                when (tag) {
+                    1 -> name = ProtoAdapter.STRING.decode(reader)
+                    2 -> id = ProtoAdapter.INT32.decode(reader)
+                    3 -> email = ProtoAdapter.STRING.decode(reader)
+                    4 -> phone.add(PhoneNumber.ADAPTER.decode(reader))
+                    else -> UnknownFieldsBuilder.UNKNOWN_FIELD
+                }
+            }
+            return Person(
+                name = name ?: throw Internal.missingRequiredFields(name, "name"),
+                id = id ?: throw Internal.missingRequiredFields(id, "id"),
+                email = email,
+                phone = phone,
+                unknownFields = unknownFields
+            )
+        }
+    }
+
+    enum class PhoneType(private val value: Int) : WireEnum {
+        MOBILE(0),
+
+        HOME(1),
+
+        /**
+         * Could be phone or fax. */
+        WORK(2);
+
+        override fun getValue(): Int = value
+
+        object ADAPTER : EnumAdapter<PhoneType>(PhoneType::class.java) {
+            override fun fromValue(value: Int): PhoneType? = values().find { it.value == value }
```

Comment:
nit: `find()` is O(N) and we probably want something closer to `O(1)` like a switch

Reply:
totally fine for now